packages/salesforcedx-vscode-core/src/commands/forceAuthWebLogin.ts

```diff
@@ -102,8 +108,31 @@ export class ForceAuthWebLoginDemoModeExecutor extends ForceAuthDemoModeExecutor
   }
 }
 
-export class AliasGatherer implements ParametersGatherer<Alias> {
-  public async gather(): Promise<CancelResponse | ContinueResponse<Alias>> {
+export class AuthParamsGatherer implements ParametersGatherer<AuthParams> {
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<AuthParams>
+  > {
+    const orgTypeItems: vscode.QuickPickItem[] = [
+      { label: 'Production', detail: 'login.salesforce.com' },
+      { label: 'Sandbox', detail: 'test.salesforce.com' },
+      { label: 'Custom', detail: 'Use a custom login URL' }
+    ];
+    const selection = await vscode.window.showQuickPick(orgTypeItems);
+    const orgType = (selection as vscode.QuickPickItem).label;
+    let loginUrl: string | undefined;
+    if (orgType === 'Custom') {
```

Comment:
I think we should validate that what the user enters is a `url` or else the command just hangs and the user doesn't receive any error. We might also want to ask the cli team about it cause I never got an error back from the cli when using an invalid url.

Reply:
New Salesforce CLI plug-ins built on oclif can validate URLs. https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_plugins.meta/sfdx_cli_plugins/cli_plugins_customize_parameters_types.htm 

Not sure that helps us at the moment, though.

Reply:
@lcampos I could be wrong, but this doesn't seem like something we'd have to implement if the CLI handled it as expected by throwing an error. If we still want to implement the validation though I can do so.

Reply:
@brpowell yeah, I'll log a bug on the cli and let's keep this as is.